compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1163,6 +1163,426 @@ TR::Register *OMR::X86::TreeEvaluator::cstoreEvaluator(TR::Node *node, TR::CodeG
 
 // icstoreEvaluator handled by cstoreEvaluator
 
+static TR::Register *REPArraycmpEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register *resultRegister = cg->allocateRegister(TR_GPR);
+   TR::Node *src1AddrNode = node->getChild(0);
+   TR::Node *src2AddrNode = node->getChild(1);
+   TR::Node *lengthNode = node->getChild(2);
+   
+   int32_t byteLen = 4;
+   if (TR::Compiler->target.is64Bit())
+      byteLen = 8;
+
+   TR::Register *tempSrc1AddrRegister = cg->evaluate(src1AddrNode);;
+   TR::Register *tempSrc2AddrRegister = cg->evaluate(src2AddrNode);
+   TR::Register *tempArrayLenRegister = cg->evaluate(lengthNode);
+
+   TR::Register *src1AddrRegister = cg->allocateRegister();
+   TR::Register *src2AddrRegister = cg->allocateRegister();
+   TR::Register *arrayLenRegister = cg->allocateRegister();
+   TR::Register *counterRegister = cg->allocateRegister();
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *notEqualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *lessThanLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endResultLabel = generateLabelSymbol(cg);
+   
+   TR::LabelSymbol *wordCompareLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *dWordCompareLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *qWordCompareLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *equalLabel = generateLabelSymbol(cg);
+
+   //Move into registers to for use
```

Comment:
These moves are done regardless of the lifetime of the original register (i.e., if the ref count is one you don't need to copy it).  I think a `gprClobberEvaluate` instead of an `evaluate` on these nodes is what you want, as you'll get the copy done for you.